src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -122,21 +142,6 @@ public function validate(mixed $entity, Constraint $constraint)
             return;
         }
 
-        if (null !== $constraint->entityClass) {
-            /* Retrieve repository from given entity name.
-             * We ensure the retrieved repository can handle the entity
-             * by checking the entity is the same, or subclass of the supported entity.
-             */
-            $repository = $em->getRepository($constraint->entityClass);
-            $supportedClass = $repository->getClassName();
-
-            if (!$entity instanceof $supportedClass) {
-                throw new ConstraintDefinitionException(sprintf('The "%s" entity repository does not support the "%s" entity. The entity should be an instance of or extend "%s".', $constraint->entityClass, $class->getName(), $supportedClass));
-            }
-        } else {
-            $repository = $em->getRepository($entity::class);
-        }
-
```

Comment:
I'd still like you to move this block BACK down here. And remove `$repository =` from https://github.com/symfony/symfony/pull/38662/files#diff-fd1785451fbb1d6abe41ae644ddbe7b39b6f03f3ba31f7ce07a18ef376e61293R88 and, instead, keep it in the `else` like we have with this block.

Reply:
Done